src/TTextEdit.cpp

```diff
@@ -1706,18 +1713,51 @@ void TTextEdit::resizeEvent(QResizeEvent* event)
 
 void TTextEdit::wheelEvent(QWheelEvent* e)
 {
-    int k = 3;
-    if (e->delta() < 0) {
-        mpConsole->scrollDown(abs(k));
-        e->accept();
-        return;
-    }
-    if (e->delta() > 0) {
-        mpConsole->scrollUp(k);
-        e->accept();
-        return;
-    }
-    e->ignore();
+    // Make the speed up be half of the (upper pane) lines - need to round it so
+    // that a decimal part does not make the end +/- value for up/down different
+    // in magitude:
+    double ySpeedUp = qRound(mpConsole->mUpperPane->getScreenHeight() / 2.0);
+    // Just a number plucked out of the air for the x-direction:
+    double xSpeedUp = 10.0;
+
+    QPointF delta = e->angleDelta();
+    // Convert to degrees:
+    delta /= 8.0;
+    // Allow the control key to introduce a speed up!:
+    delta.rx() *= (e->modifiers() & Qt::ControlModifier ? xSpeedUp : 1.0);
+    delta.ry() *= (e->modifiers() & Qt::ControlModifier ? ySpeedUp : 1.0);
+    // Add on any previously stored (integer) remainder:
+    delta += mMouseWheelRemainder;
+    // Convert to 15 degree steps and record them:
+    int xDelta = qRound(delta.x() / 15.0);
+    int yDelta = qRound(delta.y() / 15.0);
+    // Store the (rounded) remainder
+    mMouseWheelRemainder = QPoint(delta.x() - (15 * xDelta), delta.y() - (15 * yDelta));
+
+    qDebug().noquote().nospace() << "TTextEdit::wheelEvent(...) INFO - in \""
```

Comment:
This can be removed, right?

Reply:
W.I.P. on the horizontal scrolling aspect at the moment and I am using that. Mayhaps I should move that to a separate PR...

Reply:
Agree, another PR is good